.github/workflows/CD.yml

```diff
@@ -0,0 +1,116 @@
+name: CD
+
+on:
+  push:
+    tags:        
+      - '**' 
+
+jobs:
+  validate-tag:
+    name: Validate tag
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+        with:
+          fetch-tags: true
+          fetch-depth: 0
+      
+      - name: Check no jumps in tags
+        run: |
+          # If a previous tag is X.Y, valid tags are only X.(Y+1) or (X+1).0
+          if [[ $(git tag -l | wc -l) != 1 ]]; then # If there is only one tag, no need to check
+            prev=($(git tag -l --sort=-v:refname | head -n 2 | tail -n 1 | tr "." " "))
+            val1="${prev[0]}.$((${prev[1]}+1))" # X.(Y+1)
+            val2="$((${prev[0]}+1)).0" # (X+1).0
+            if [[ "${{github.ref_name}}" != "$val1" ]] && \
+            [[ "${{github.ref_name}}" != "$val2" ]]; then
+                echo "Invalid tag \"${{github.ref_name}}\" when previous tag is 
+                \"$(sed 's/ /./' <<< ${prev[@]})\"."
```

Comment:
It might be simpler to have the previous tag (before the `tr` stuff)  as a variable and just putting that in the echo rather than using command substitution. 

Something like:
```
prev_tag=$(git tag -l --sort=-v:refname | head -n 2 | tail -n 1)
prev_tag_components=($($prev_tag | tr  "." " "))  # or something like this
```

Reply:
I refactored for clarity in [210e00add46f0c74d626e77b540e276a4d8b6609](https://github.com/ACCESS-NRI/replace_landsurface/pull/45/commits/210e00add46f0c74d626e77b540e276a4d8b6609)